public/js/menu.js

```diff
@@ -1,66 +1,67 @@
-        };
-    });
+function updateFilterNumbers() {
+    const form = document.querySelector("#filter-options form");
+    const formData = new FormData(form);
+    const queryString = new URLSearchParams(formData).toString();
 
-    function updateFilterNumbers() {
-        const form = document.querySelector('#filter-options form');
-        const formData = new FormData(form);
-        const queryString = new URLSearchParams(formData).toString();
-        fetch(
-            `https://cmdmethods.onrender.com/api/filter-numbers?${queryString}`
-        )
-            .then((response) => response.json())
-            .then((filterNumbers) => {
-                const list = document.querySelectorAll('.filterNumber');
-                list.forEach((el) => {
-                    if (filterNumbers[el.dataset.filter] != null) {
-                        if (filterNumbers[el.dataset.filter] === 0) {
-                            el.parentElement.previousElementSibling.disabled = true;
-                        } else {
-                            el.parentElement.previousElementSibling.disabled = false;
-                        }
-                        el.textContent = `(${
-                            filterNumbers[el.dataset.filter]
-                        })`;
+    fetch(`/api/filter-numbers?${queryString}`)
+        .then((response) => response.json())
+        .then((filterNumbers) => {
+            const list = document.querySelectorAll(".filterNumber");
+            list.forEach((el) => {
+                if (filterNumbers[el.dataset.filter] != null) {
+                    if (filterNumbers[el.dataset.filter] === 0) {
+                        el.parentElement.previousElementSibling.disabled = true;
+                    } else {
+                        el.parentElement.previousElementSibling.disabled = false;
                     }
-                });
-                document.querySelector(
-                    '#filter-options input[type=submit]'
-                ).value = `Show methods (${filterNumbers.total})`;
-            })
-            .catch((error) => console.log(error));
-    }
+                    el.textContent = `(${filterNumbers[el.dataset.filter]})`;
+                }
+            });
+            document.querySelector(
+                "#filter-options input[type=submit]"
+            ).value = `Show methods (${filterNumbers.total})`;
+        })
+        .catch((error) => console.log(error));
+}
 
-    updateFilterNumbers();
+updateFilterNumbers();
 
-    window.onscroll = function () {
-        const categoryLogos = document.querySelectorAll('.flipable');
-        categoryLogos.forEach((element) => {
-            const observer = new IntersectionObserver(
-                (entries) => {
-                    element.classList.toggle('flip', entries[0].isIntersecting);
-                },
-                { threshold: 1 }
-            );
-            observer.observe(element);
+const observer = new IntersectionObserver(
```

Comment:
Instead of creating a new observer for every scroll event, we'll create a single observer that'll be used for the watched elements

Reply:
oeh, ja goeie. Dat was erg inefficient